utils/priority_mutex_map_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/sync/errgroup"
+)
+
+func TestMutexMap(t *testing.T) {
+	arr := []string{}
+	m := NewMutexMap()
+	g, _ := errgroup.WithContext(context.Background())
+
+	// Lock while adding all locks
+	m.GLock()
+
+	// Add another GLock
+	g.Go(func() error {
+		m.GLock()
+		arr = append(arr, "global")
```

Comment:
Is it worth using a different identifier, e.g. `globalB` and `globalA` to more clearly disambiguate which is ocurring first?